### PR TITLE
Version 6.0.0: AutoMapper 11 support

### PR DESCRIPTION
**Pull request by Pierta**
Moved to .NET Standard 2.1; replaced AutoMapper 10 with 11

### CODE REVIEW

**Comment by Pierta** [comment]
Looks like appveyor pipeline should be updated as well

**Comment by Pierta** [comment]
@lbargaoanu, hello! Can you, pls, review this change? And help with a failed pipeline :)

File: AutoMapper.Data.Tests/DataReaderMapping.cs [hunk 1]
```diff
@@ -123,8 +124,8 @@ public class When_mapping_a_data_reader_to_matching_dtos
         public When_mapping_a_data_reader_to_matching_dtos()
         {
              MapperConfiguration configuration = new MapperConfiguration(cfg =>{
-                 cfg.Mappers.Insert(0, new DataReaderMapper());
-                 cfg.AddMemberConfiguration().AddMember<DataRecordMemberConfiguration>();
+                 InternalApi.Internal(cfg).Mappers.Insert(0, new DataReaderMapper());
```
Comment (lbargaoanu) [inline AutoMapper.Data.Tests/DataReaderMapping.cs:127]: Rebase and then use the extension method everywhere. It looks better :)

File: AutoMapper.Data/AutoMapper.Data.csproj [hunk 2]
```diff
@@ -16,10 +16,10 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="AutoMapper" Version="[10.0.0, 11.0.0)" />
+    <PackageReference Include="AutoMapper" Version="[11.0.0, 12.0.0)" />
   </ItemGroup>
 
-  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.1' ">
```
Comment (lbargaoanu) [inline AutoMapper.Data/AutoMapper.Data.csproj:22]: The condition is not needed anymore, so move these in the group above.

File: AutoMapper.Data/ConfigurationExtensions.cs [hunk 3]
```diff
@@ -10,8 +11,8 @@ public static void AddDataReaderMapping(this IMapperConfigurationExpression conf
 
         public static void AddDataReaderMapping(this IMapperConfigurationExpression configuration, bool enableYieldReturn)
         {
-            configuration.Mappers.Insert(0, new DataReaderMapper { YieldReturnEnabled = enableYieldReturn });
-            configuration.AddMemberConfiguration().AddMember<DataRecordMemberConfiguration>();
+            InternalApi.Internal(configuration).Mappers.Insert(0, new DataReaderMapper { YieldReturnEnabled = enableYieldReturn });
```
Comment (lbargaoanu) [inline AutoMapper.Data/ConfigurationExtensions.cs:14]: `AddDataReaderMapping()` or the extension method `Internal()`, _everywhere_ :)

Comment (Pierta) [reply AutoMapper.Data/ConfigurationExtensions.cs:14]: Ugh, yeah, thanks for the notice

Comment (Pierta) [reply AutoMapper.Data/ConfigurationExtensions.cs:14]: Checked all the places, but didn't find any gaps. May be I didn't get your point correctly 😕
Everywhere either `AddDataReaderMapping()` or the extension method `Internal()` is used.

File: AutoMapper.Data.Tests/DataReaderMappingTests.cs [hunk 4]
```diff
@@ -188,7 +188,7 @@ public class When_mapping_a_data_reader_to_a_dto_and_the_map_does_not_exist
     {
         public When_mapping_a_data_reader_to_a_dto_and_the_map_does_not_exist()
         {
-            MapperConfiguration configuration = new MapperConfiguration(cfg => cfg.Mappers.Insert(0, new DataReaderMapper()));
+            MapperConfiguration configuration = new MapperConfiguration(cfg => InternalApi.Internal(cfg).Mappers.Insert(0, new DataReaderMapper()));
```
Comment (lbargaoanu) [inline AutoMapper.Data.Tests/DataReaderMappingTests.cs:191]: `cfg.Internal()...`

File: AutoMapper.Data/ConfigurationExtensions.cs [hunk 5]
```diff
@@ -1,27 +1,34 @@
...
 
-        public static void AddDataReaderProfile(this IMapperConfigurationExpression configuration, Profile profile)
+        public static void AddDataRecordMember(this Profile profile)
         {
-            configuration.AddDataReaderMapping();
-            profile.AddDataRecordMember();
-            configuration.AddProfile(profile);
+            if (profile != null)
+            {
+                InternalApi.Internal(profile).AddMemberConfiguration().AddMember<DataRecordMemberConfiguration>();
```
[the remark this answers is not in the document]
Comment (Pierta) [reply AutoMapper.Data/ConfigurationExtensions.cs:30]: :) ok, got it, finally. Thanks

File: AutoMapper.Data/Mappers/DataReaderMapper.cs [hunk 6]
```diff
@@ -21,19 +22,20 @@ public class DataReaderMapper : IObjectMapper
...
             if (IsDataReader(sourceExpression.Type, destExpression.Type))
             {
                 ParameterExpression itemParam;
                 Expression itemExpr;
 
                 try
-                {
-                    itemExpr = CollectionMapperExpressionFactory.MapItemExpr(configurationProvider, profileMap, typeof(IEnumerable<IDataRecord>), destExpression.Type, contextExpression, out itemParam);
+                {
+                    itemExpr = InternalApi.Internal(configurationProvider).MapItemExpr(profileMap, memberMap,
```
Comment (lbargaoanu) [inline AutoMapper.Data/Mappers/DataReaderMapper.cs:37]: I think `Internal` is not needed here.

File: AutoMapper.Data/ConfigurationExtensions.cs (hunk 3 above)
Comment (Pierta) [reply AutoMapper.Data/ConfigurationExtensions.cs]: Fixed!

File: AutoMapper.Data.Tests/DataReaderMappingTests.cs (hunk 4 above)
Comment (Pierta) [reply AutoMapper.Data.Tests/DataReaderMappingTests.cs]: Fixed!

File: AutoMapper.Data/Mappers/DataReaderMapper.cs (hunk 6 above)
Comment (Pierta) [reply AutoMapper.Data/Mappers/DataReaderMapper.cs]: Yes, you're right. Fixed!

**Comment by lbargaoanu** [comment]
Thanks!

**Pull request merged**